controllers/space/space_controller_test.go

```diff
@@ -0,0 +1,796 @@
+				space := spacetest.NewSpace("oddity", basicTier.Name,
+					spacetest.WithTargetCluster("member-1"),
+					spacetest.WithFinalizer())
+				hostClient := test.NewFakeClient(t, space, basicTier)
+				member1Client := test.NewFakeClient(t)
+				member1Client.MockCreate = func(ctx context.Context, obj client.Object, opts ...client.CreateOption) error {
+					if _, ok := obj.(*toolchainv1alpha1.NSTemplateSet); ok {
+						return fmt.Errorf("mock error!")
+					}
+					return member1Client.Client.Create(ctx, obj, opts...)
+				}
+				member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+				member2 := NewMemberCluster(t, "member-2", corev1.ConditionTrue)
+				ctrl := newReconciler(hostClient, member1, member2)
+
+				// when
+				res, err := ctrl.Reconcile(context.TODO(), reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Namespace: space.Namespace,
+						Name:      space.Name,
+					},
+				})
+
+				// then
+				require.EqualError(t, err, "mock error!")
+				assert.Equal(t, reconcile.Result{Requeue: false}, res)
+				spacetest.AssertThatSpace(t, space.Namespace, space.Name, hostClient).
+					HasStatusTargetCluster("member-1").
+					HasConditions(toolchainv1alpha1.Condition{
+						Type:    toolchainv1alpha1.ConditionReady,
+						Status:  corev1.ConditionFalse,
+						Reason:  toolchainv1alpha1.SpaceUnableToCreateNSTemplateSetReason,
+						Message: "mock error!",
+					})
+			})
+
+			t.Run("error while updating status after creating NSTemplateSet", func(t *testing.T) {
+				// given
+				space := spacetest.NewSpace("oddity", basicTier.Name,
+					spacetest.WithTargetCluster("member-1"),
+					spacetest.WithFinalizer())
+				hostClient := test.NewFakeClient(t, space, basicTier)
+				hostClient.MockStatusUpdate = func(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
+					if _, ok := obj.(*toolchainv1alpha1.Space); ok {
+						return fmt.Errorf("mock error!")
+					}
+					return hostClient.Client.Status().Update(ctx, obj, opts...)
+				}
+				member1 := NewMemberCluster(t, "member-1", corev1.ConditionTrue)
+				member2 := NewMemberCluster(t, "member-2", corev1.ConditionTrue)
+				ctrl := newReconciler(hostClient, member1, member2)
+
+				// when
+				res, err := ctrl.Reconcile(context.TODO(), reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Namespace: space.Namespace,
+						Name:      space.Name,
+					},
+				})
+
+				// then
+				require.EqualError(t, err, "mock error!")
+				assert.Equal(t, reconcile.Result{Requeue: false}, res)
+				spacetest.AssertThatSpace(t, space.Namespace, space.Name, hostClient).
+					HasStatusTargetCluster(""). // not set
+					HasNoConditions()           // not set
+			})
+		})
+	})
+
+	t.Run("delete", func(t *testing.T) {
+
+		t.Run("success", func(t *testing.T) {
+
+			t.Run("after space was successfully provisioned", func(t *testing.T) {
+
+				t.Run("when space is deleted", func(t *testing.T) {
```

Comment:
Do we really need this amount of subtests when there is no shared common setup? 
In addition. It would be nice to start a brand new test for the new use-case - deletion. It's hard to follow what setup is reused.

Reply:
I like it that way, but if it's really an issue, then I'm fine with splitting the test.

Reply:
Please split it at least into two tests:
1. provisioning
2. deleting

and please avoid so many subtests without using any common setup - it's really hard to review it. 

Just out of curiosity - what is the benefit of having such 4 subtests.

Reply:
just a matter of organizing the tests, but if you prefer otherwise, then fine

Reply:
done in 87b2b42